packages/atomic/src/utils/local-storage-utils.tsx

```diff
@@ -0,0 +1,108 @@
+import {SearchEngine} from '@coveo/headless';
+
+export enum StorageItems {
+  RECENT_QUERIES = 'coveo-recent-queries',
+}
+
+export class SafeStorage implements Storage {
+  constructor(private logger: SearchEngine['logger']) {}
```

Comment:
Should we use the engine logger which might be silenced? I feel for environment errors, we can log using `console`?